### PR TITLE
Evaluate tts

### PR DESCRIPTION
**Pull request by KariSt1**
Added tts evaluation code from tts-evaluate branch on GreynirServer. The evaluation creates audio files of text snippets in all available Icelandic voices and plays them for the user. Each audio file can then be rated by the user for correctness and naturalness.

### CODE REVIEW

**Review by sultur — COMMENTED** [review]
lgtm, small nitpicks, we also need to fix the weird test failures.

File: tools/evaluate_tts.py [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+
+tts_textar_file = Path(__file__).with_name("tts_textar.txt")
+tts_ratings_file = Path(__file__).with_name("tts_ratings.tsv")
+
+app = typer.Typer()
+
+
+def _die(msg: str, exit_code: int = 1) -> None:
+    print(msg, file=sys.stderr)
+    sys.exit(exit_code)
```
[the remark this answers is not in the document]
Comment (KariSt1) [reply tools/evaluate_tts.py]: Done

File: tools/evaluate_tts.py [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+
+
+def read_voices() -> list[str]:
+    """Read the voices to be evaluated from a file."""
+    voices: list[str] = []
+    for key, value in VOICES.items():
+        if value["lang"] == "is-IS":
+            voices.append(key)
+    return voices
```
[the remark this answers is not in the document]
Comment (KariSt1) [reply tools/evaluate_tts.py]: Removed function and replaced it with the oneliner.

File: tools/evaluate_tts.py [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+    voices = read_voices()
+    if not voices:
+        _die("No voices to evaluate.")
+    texts = read_textar()
+    if not texts:
+        _die("No text snippets to evaluate.")
+    # read the ratings file
+    ratings = {}
+    try:
+        with open(tts_ratings_file) as f:
```
[the remark this answers is not in the document]
Comment (KariSt1) [reply tools/evaluate_tts.py]: Changed all occurrences to `tts_ratings_file.open()`

File: tools/evaluate_tts.py [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+
+def read_textar() -> list[str]:
+    """Read the text snippets to be evaluated from a file."""
+    textar: list[str] = []
+    with open(tts_textar_file) as f:
+        for line in f:
+            line = line.strip()
+            if line:
+                textar.append(line)
+    return textar
```
[the remark this answers is not in the document]
Comment (KariSt1) [reply tools/evaluate_tts.py]: Renamed the variable and replaced the function with the one liner.

File: tools/evaluate_tts.py [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+be resumed at a later time.
+"""
+
+from typing import Any
+
+import sys
+from pathlib import Path
+
+from icespeak import VOICES, TTSOptions, tts_to_file
+from icespeak.cli import _play_audio_file
```
[the remark this answers is not in the document]
Comment (KariSt1) [reply tools/evaluate_tts.py]: `play_audio_file` is now a public function in `cli.py`.

File: tools/evaluate_tts.py [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+be resumed at a later time.
+"""
+
+from typing import Any
+
+import sys
+from pathlib import Path
+
+from icespeak import VOICES, TTSOptions, tts_to_file
+from icespeak.cli import play_audio_file
```
Comment (sveinbjornt) [inline tools/evaluate_tts.py:34]: If this function is being used outside the cli, it should probably find a new home.

**Pull request merged**